cpp/src/io/comp/compression.cpp

```diff
@@ -325,73 +327,100 @@ void host_compress(compression_type compression,
                    device_span<codec_exec_result> results,
                    rmm::cuda_stream_view stream)
 {
-  if (compression == compression_type::NONE) { return; }
+  CUDF_FUNC_RANGE();
+  if (compression == compression_type::NONE or inputs.empty()) { return; }
 
   auto const num_chunks = inputs.size();
   auto const h_inputs   = cudf::detail::make_host_vector_async(inputs, stream);
   auto const h_outputs  = cudf::detail::make_host_vector_async(outputs, stream);
   stream.synchronize();
 
-  // Generate order vector to submit largest tasks first
-  std::vector<size_t> task_order(num_chunks);
-  std::iota(task_order.begin(), task_order.end(), 0);
-  std::sort(task_order.begin(), task_order.end(), [&](size_t a, size_t b) {
-    return h_inputs[a].size() > h_inputs[b].size();
-  });
```

Comment:
removed this sorting because we now sort inputs regardless of the (de)comp engine.